integrationTests/testInitializer.go

```diff
@@ -672,29 +673,36 @@ func ProposeAndSyncBlocks(
 	}
 
 	if nodes[0].ShardCoordinator.NumberOfShards() == 1 {
-		return round
+		return round, nonce
 	}
 
 	// cross shard smart contract call is first processed at sender shard, notarized by metachain, processed at
 	// shard with smart contract, smart contract result is notarized by metachain, then finally processed at the
 	// sender shard
 	numberToPropagateToEveryShard := 5
 	for i := 0; i < numberToPropagateToEveryShard; i++ {
-		round = ProposeAndSyncOneBlock(t, nodes, idxProposers, round)
+		round, nonce = ProposeAndSyncOneBlock(t, nodes, idxProposers, round, nonce)
 	}
 
-	return round
+	return round, nonce
 }
 
 func ProposeAndSyncOneBlock(
 	t *testing.T,
 	nodes []*TestProcessorNode,
 	idxProposers []int,
 	round uint64,
-) uint64 {
-	ProposeBlock(nodes, idxProposers, round)
+	nonce uint64,
+) (uint64, uint64) {
+	ProposeBlock(nodes, idxProposers, round, nonce)
 	SyncBlock(t, nodes, idxProposers, round)
 	round = IncrementAndPrintRound(round)
+	nonce = IncrementNonce(nonce)
 
-	return round
+	return round, nonce
+}
+
+func IncrementNonce(nonce uint64) uint64 {
```

Comment:
I don't think we need a function for this one

Reply:
ok